job_executor/app.py

```diff
@@ -62,75 +62,76 @@ def fix_interrupted_jobs():
     logger.info(f"Found {len(interrupted_jobs)} interrupted jobs")
 
     for job in interrupted_jobs:
-        job_operation = job.parameters.operation
-        logger.info(
-            f"{job.job_id}: Rolling back job with operation "
-            f'"{job_operation}"'
-        )
-        if job_operation in ["ADD", "CHANGE", "PATCH_METADATA"]:
-            if job.status == "importing":
-                rollback.rollback_manager_phase_import_job(
-                    job.job_id, job_operation, job.parameters.target
-                )
-                logger.info(
-                    f"{job.job_id}: Rolled back importing of job with "
-                    f'operation "{job_operation}". Retrying from status '
-                    '"built"'
-                )
-                job_service.update_job_status(
-                    job.job_id,
-                    "built",
-                    "Reset to built status will be due to "
-                    "unexpected interruption",
-                )
-            else:
-                rollback.rollback_worker_phase_import_job(
-                    job.job_id, job_operation, job.parameters.target
-                )
-                logger.info(
-                    f'{job.job_id}: Setting status to "failed" for '
-                    f"interrupted job"
-                )
-                job_service.update_job_status(
-                    job.job_id,
-                    "failed",
-                    "Job was failed due to an unexpected interruption",
-                )
-        elif job_operation in ["SET_STATUS", "DELETE_DRAFT", "REMOVE"]:
+        fix_interrupted_job(job)
+
+
+def fix_interrupted_job(job):
+    job_operation = job.parameters.operation
+    logger.info(
+        f"{job.job_id}: Rolling back job with operation " f'"{job_operation}"'
+    )
+    if job_operation in ["ADD", "CHANGE", "PATCH_METADATA"]:
+        if job.status == "importing":
+            rollback.rollback_manager_phase_import_job(
+                job.job_id, job_operation, job.parameters.target
+            )
             logger.info(
-                'Setting status to "queued" for '
-                f"interrupted job with id {job.job_id}"
+                f"{job.job_id}: Rolled back importing of job with "
+                f'operation "{job_operation}". Retrying from status '
+                '"built"'
```

Comment:
A nitpick, Forgot I had pyright on and saw it complain about "built" not being a valid status however it is a valid based on our pydantic model.

Pyright wants to use enum members: JobStatus.BUILT. This provides better typesafety(typosafety) and if we are considering pyright as a typechecker in the future these are small fixes we can do along the way.

Reply:
Yes, true. I see that Pylance is reporting this as well as some other problems. I suggest we fix those in a separate task (and we should standarize on a type checker across our projects).

Reply:
agreed! (and pylance uses pyright under the hood so both of you getting the same errors makes sense)